TreeMaker/python/makeTreeFromMiniAOD_cff.py

```diff
@@ -1115,6 +1117,52 @@ def makeTreeFromMiniAOD(self,process):
                 'HiddenSector:isHV(JetsAK8_isHV)'
             ])
 
+    ## ----------------------------------------------------------------------------------------------
+    ## Emerging jets
+    ## ----------------------------------------------------------------------------------------------
+    if self.emerging:
+        process.load("TrackingTools/TransientTrack/TransientTrackBuilder_cfi")
+
+        from TreeMaker.Utils.candidateTrackMaker_cfi import candidateTrackFilter
+        process.trackFilter = candidateTrackFilter.clone(
+                vertexInputTag    = cms.InputTag("goodVertices"),
+                pfCandidatesTag   = cms.InputTag("packedPFCandidates"),
+                lostTracksTag     = cms.InputTag("lostTracks"),
+        )
+        self.VectorXYZVector.extend(['trackFilter:trks(Tracks)'])
+        self.VectorXYZPoint.extend(['trackFilter:trksreferencepoint(Tracks_referencePoint)'])
+        self.VectorBool.extend(['trackFilter:trksmatchedtopfcand(Tracks_matchedToPFCandidate)',])
+        self.VectorDouble.extend([
+            'trackFilter:trksdzpv(Tracks_dzPV0)',
+            'trackFilter:trksdzerrorpv(Tracks_dzErrorPV0)',
+            'trackFilter:trksdxypv(Tracks_dxyPV0)',
+            'trackFilter:trksdxyerrorpv(Tracks_dxyErrorPV0)',
+            'trackFilter:trksnormalizedchi2(Tracks_normalizedChi2)',
+            'trackFilter:trkspterror(Tracks_ptError)',
+            'trackFilter:trksetaerror(Tracks_etaError)',
+            'trackFilter:trksphierror(Tracks_phiError)',
+            'trackFilter:trksqoverperror(Tracks_qoverpError)',
+            'trackFilter:trksip2d(Tracks_IP2DPV0)',
+            'trackFilter:trksip2dsig(Tracks_IP2dSigPV0)',
+            'trackFilter:trksip3d(Tracks_IP3DPV0)',
+            'trackFilter:trksip3dsig(Tracks_IP3DSigPV0)',
+            'trackFilter:pfcandsdzassociatedpv(PFCandidates_dzAssociatedPV)',
```

Comment:
This saves some quantities associated with PFCandidates (distinct from tracks), but doesn't save the PFCandidate 4-vectors. Would those also be needed? (Leading to another large space increase...)

Reply:
Technically the pT of the PFCandidate doesn't need to match that of the track, but according to the MiniAOD twiki that is rare. Most often the information is the same, so the only thing you would be adding is an energy measurement. Do you think this is needed?

Reply:
So the vectors starting with "Track_" and the vectors starting with "PFCandidate_" are supposed to be the same size? This breaks our naming convention.

Reply:
They should be the same size by construction. How does this break the naming convention?

Reply:
Branches with different names are not expected to have the same size.

Reply:
Fixed